### PR TITLE
Add HostInfo to ImprovedStreamsConfig

### PR DESCRIPTION
**Pull request by raminqaf**

### CODE REVIEW

File: streams-bootstrap-core/src/main/java/com/bakdata/kafka/RunningStreams.java
Comment (raminqaf) [inline streams-bootstrap-core/src/main/java/com/bakdata/kafka/RunningStreams.java:53]: TODO: tests

Comment (philipp94831) [reply streams-bootstrap-core/src/main/java/com/bakdata/kafka/RunningStreams.java:53]: Alternatively we cann add the method to https://github.com/bakdata/streams-bootstrap/blob/9ce7863247d9795b66f06bfc048d5bcfe0122197/streams-bootstrap-core/src/main/java/com/bakdata/kafka/ImprovedStreamsConfig.java and change the field here to ImprovedStreamsConfig

Comment (raminqaf) [reply streams-bootstrap-core/src/main/java/com/bakdata/kafka/RunningStreams.java:53]: hmm... not a bad idea, but I don't like that we would have two fields (`StreamsConfig` and `ImprovedStreamsConfig`) in the same class. This would be a bit confusing. Or do you mean to change the field type of `StreamsConfig` to `ImprovedStreamsConfig`?

Comment (philipp94831) [reply streams-bootstrap-core/src/main/java/com/bakdata/kafka/RunningStreams.java]: Change the field time

File: streams-bootstrap-core/src/main/java/com/bakdata/kafka/ImprovedStreamsConfig.java
Comment (philipp94831) [inline streams-bootstrap-core/src/main/java/com/bakdata/kafka/ImprovedStreamsConfig.java:78]:
```suggestion
    public Optional<HostInfo> getApplicationServer() {
```

**Review by philipp94831 — CHANGES_REQUESTED** [review]

**Comment by philipp94831** [comment]
Lets merge this once #207 is merged

**Comment by philipp94831** [comment]
Please merge v3 branch

File: streams-bootstrap-core/src/test/java/com/bakdata/kafka/ImprovedStreamsConfigTest.java
Comment (philipp94831) [inline streams-bootstrap-core/src/test/java/com/bakdata/kafka/ImprovedStreamsConfigTest.java:100]:
```suggestion
    void shouldHaveHostInfoIfApplicationServiceIsConfigured() {
```

File: streams-bootstrap-core/src/test/java/com/bakdata/kafka/ImprovedStreamsConfigTest.java
Comment (philipp94831) [inline streams-bootstrap-core/src/test/java/com/bakdata/kafka/ImprovedStreamsConfigTest.java:110]:
```suggestion
    void shouldReturnEmptyHostInfoIfApplicationServiceIsNotConfigured() {
```

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: streams-bootstrap-core/src/test/java/com/bakdata/kafka/ImprovedStreamsConfigTest.java
Comment (philipp94831) [inline streams-bootstrap-core/src/test/java/com/bakdata/kafka/ImprovedStreamsConfigTest.java:115]:
```suggestion
                .isNotPresent();
```

**Review by philipp94831 — APPROVED** [review]

**Pull request merged**
